src/forms/medication-dispense-review.component.tsx

```diff
@@ -489,6 +500,52 @@ const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
             });
           }}
         />
+
+        <OpenmrsDatePicker
+          id="dispenseDate"
+          labelText={t('dispenseDate', 'Date of Dispense')}
+          minDate={prescriptionDate ? dayjs(prescriptionDate).startOf('day').toDate() : null}
+          maxDate={dayjs().toDate()}
+          onChange={(selectedDate) => {
+            const currentDate = dayjs(medicationDispense.whenHandedOver);
+            updateMedicationDispense({
+              ...medicationDispense,
+              whenHandedOver: isSameDay(currentDate, selectedDate)
+                ? currentDate.toISOString()
+                : selectedDate.toString(), // to preserve any time component, only update if the day actually changes
+            });
+          }}
+          value={dayjs(medicationDispense.whenHandedOver).startOf('day').toDate()}></OpenmrsDatePicker>
```

Comment:
The time (12:00AM) being displayed might be because you added `startOf('day')` here